include/mata/nfa/builder.hh

```diff
@@ -44,26 +48,19 @@ Nfa create_sigma_star_nfa(Alphabet* alphabet = new OnTheFlyAlphabet{});
 
 /** Loads an automaton from Parsed object */
 // TODO this function should the same thing as the one taking IntermediateAut or be deleted
-Nfa construct(const Mata::Parser::ParsedSection& parsec, Alphabet* alphabet, StringToStateMap* state_map = nullptr);
+Nfa construct(const Mata::Parser::ParsedSection& parsec, Alphabet* alphabet, StateNameValueMap* state_map = nullptr);
 
 /** Loads an automaton from Parsed object */
-Nfa construct(const Mata::IntermediateAut& inter_aut, Alphabet* alphabet, StringToStateMap* state_map = nullptr);
+Nfa construct(const Mata::IntermediateAut& inter_aut, Alphabet* alphabet, StateNameValueMap* state_map = nullptr);
 
 template<class ParsedObject>
-Nfa construct(const ParsedObject& parsed, Mata::StringToSymbolMap* symbol_map = nullptr,
-              StringToStateMap* state_map = nullptr) {
-    Mata::StringToSymbolMap tmp_symbol_map;
-    if (symbol_map) {
-        tmp_symbol_map = *symbol_map;
-    }
-    Mata::OnTheFlyAlphabet alphabet(tmp_symbol_map);
-
-    Nfa aut = construct(parsed, &alphabet, state_map);
-
-    if (symbol_map) {
-        *symbol_map = alphabet.get_symbol_map();
+Nfa construct(const ParsedObject& parsed, Alphabet* alphabet = nullptr,
+              StateNameValueMap* state_map = nullptr) {
+    OnTheFlyAlphabet tmp_alphabet{};
+    if (!alphabet) {
+        alphabet = &tmp_alphabet;
     }
-    return aut;
+    return construct(parsed, alphabet, state_map);
```

Comment:
I really don't like the function name construct, but that is besides the point in this PR.

Reply:
I made a note about that. It will be resolved in a later PR.

Reply:
This issue created for this problem: https://github.com/VeriFIT/mata/issues/291.